### ISSUE TITLE
Pre-training on COCO using custom YOLOv8 architecture

### ISSUE THREAD

**Issue by SigurdBakkerud** [ultralytics/ultralytics#10434]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have several YOLOv8 architectures with each having a separate yaml file describing the architecture. To maximize performance on the new architecture, I want to use transfer learning on the new models. Is there a built in way to pre-train a model on COCO through ultralytics library?
Also, let's assume there is a way to pre-train the model on COCO, and I have a way to do it. What would then be the best training configurations to reproduce the same type of pre-training as the "yolov8s.pt" weights provided by ultralytics?(number of epochs, learning rate, etc..)

**Comment by glenn-jocher** [comment]
@SigurdBakkerud hello! 👋 

Pre-training your custom YOLOv8 architectures on COCO is a great approach to leverage transfer learning!

To pre-train a model on COCO, you can simply specify your architecture's YAML file and the COCO dataset in the training command. Ultralytics conveniently auto-downloads the COCO dataset the first time, making it easy to start:

```bash
yolo train data=coco.yaml model=custom-yolov8.yaml
```

For replicating the pre-training similar to the "yolov8s.pt" weights, although specifics might vary depending on your custom architecture, a good starting point based on the general practices would be:

- **Epochs:** Around 300-500 
- **Batch size:** 16 (Adjust based on your GPU memory)
- **Learning rate:** 0.01 (With a warmup period and then a cosine decay schedule)

These are rough estimates, and the exact configuration might require some tweaking. Monitoring validation metrics closely during training helps in fine-tuning these parameters for optimal performance.

Happy coding, and may your model accuracy be high! 🚀
